### ISSUE TITLE
mongooseFactory createAndSave does not return the object

### ISSUE THREAD

**Issue by mb-dev** [goodeggs/unionized#15]
To reproduce - create a mongoose factory. Call createAndSave and inspect the return value.

Expected:
The created object instance

Actual:
undefined

**Comment by demands** [comment]
I don't think this is right. [Take a look at this passing test case.](https://github.com/goodeggs/unionized/blob/master/test/mongoose.spec.coffee#L168-L186)

There must be something specific to your use case that was causing this problem.

**Comment by mb-dev** [comment]
The failed tests are when using mongoose 3.6.

I cloned unionized, installed 3.6 and ran tests:

![screen shot 2015-07-29 at 9 53 48 am](https://cloud.githubusercontent.com/assets/807807/8963966/d4b9b54a-35d7-11e5-9117-886c81656737.png)

**Comment by demands** [comment]
Aha! Thanks Moshe :)
